### ISSUE TITLE
presto-verifier-0.263.1-executable.jar broken

### ISSUE THREAD

**Issue by yingsu00** [prestodb/presto#16877]
Downloaded the jar from https://prestodb.io/docs/current/admin/verifier.html and followed the instruction to rename and chmod. 
```
$ ./verifier
Error: Invalid or corrupt jarfile ./verifier
```
The 0.245 version was known to work:
```
$ java -jar ./presto-verifier-0.245-SNAPSHOT-executable.jar
usage: verifier <command> [<args>]

The most commonly used verifier commands are:
    help     Display help information
    verify   verify
```

The problem happened between (0.245, 0.259]. I didn't test all versions in between.

**Comment by rongrong** [comment]
cc @mayankgarg1990

**Comment by ajaygeorge** [comment]
16:43 $ git bisect bad
b043970f4122b5fd242c1916dfea2bdc348ab631 is the first bad commit
commit b043970f4122b5fd242c1916dfea2bdc348ab631
Author: Arunachalam Thirupathi <athirupathi@fb.com>
Date:   Wed Feb 3 20:01:40 2021 -0800

    Upgrade fastutil to 8.5.2

    Integer dictionary encoding for DWRF format is planned.
    For efficient implementation, fastutil putIfAbsent is preferred
    and is vailable in newer versions. Upgrading hive-dwrf
    as the current version is not compatible with fastutil 8.5.2

    Depended by https://github.com/facebookexternal/presto-facebook/pull/1434

 16:44 $ git bisect log
 ```
git bisect start
# bad: [8b7f3960c69e30bff79bec35885697165686f3a4] Add flag to enable/disable undo for metastore operations
git bisect bad 8b7f3960c69e30bff79bec35885697165686f3a4
# good: [29a5e12e084c000aac873148b4b5f1055106695c] Supports complex types in reduce_agg
git bisect good 29a5e12e084c000aac873148b4b5f1055106695c
# bad: [b888504726913f8ac8e521a47f37eb33b6a7e974] Revert "Add PrestoSparkMetadataStorage"
git bisect bad b888504726913f8ac8e521a47f37eb33b6a7e974
# bad: [b888504726913f8ac8e521a47f37eb33b6a7e974] Revert "Add PrestoSparkMetadataStorage"
git bisect bad b888504726913f8ac8e521a47f37eb33b6a7e974
# good: [e360668b2cc36d0ff1b04e4e53316d6cff3620c9] Add product tests to ci.yml
git bisect good e360668b2cc36d0ff1b04e4e53316d6cff3620c9
# good: [e360668b2cc36d0ff1b04e4e53316d6cff3620c9] Add product tests to ci.yml
git bisect good e360668b2cc36d0ff1b04e4e53316d6cff3620c9
# bad: [bd933262207d87defd838b19a2a7759aa6545ae2] Implement storage based broadcast join in Presto on Spark
git bisect bad bd933262207d87defd838b19a2a7759aa6545ae2
# good: [e7a85b3eadc87703a6320d407d9c8e550b614d93] Support enums in SQL functions
git bisect good e7a85b3eadc87703a6320d407d9c8e550b614d93
# bad: [866578801d36a5fdfb03bf6dc3a882b140f42332] Stop using Travis in favor of GH actions
git bisect bad 866578801d36a5fdfb03bf6dc3a882b140f42332
# bad: [b043970f4122b5fd242c1916dfea2bdc348ab631] Upgrade fastutil to 8.5.2
git bisect bad b043970f4122b5fd242c1916dfea2bdc348ab631
# first bad commit: [b043970f4122b5fd242c1916dfea2bdc348ab631] Upgrade fastutil to 8.5.2
```

**Comment by arunthirupathi** [comment]
I will take a look at this tomorrow and get back to this. fastutil does a code gen and it generally has problem with shading, will update on this issue tomorrow.

**Comment by ajaygeorge** [comment]
@arunthirupathi that is a great find. 
This got me curious as to why presto-verifier should be using fastutil dependency in the first place since there is no apparent need for it. 
To test, I added the dependency exclusion for presto-main from where this transitive dependency is pulled in to presto-verifier's pom.xml

```
        <dependency>
            <groupId>com.facebook.presto</groupId>
            <artifactId>presto-main</artifactId>
            <exclusions>
                <exclusion>
                    <groupId>it.unimi.dsi</groupId>
                    <artifactId>fastutil</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
  ```
I was able to compile and generate the jar and it yields the working jar and as expected we are around 12k classes down.
So were just un-necessarily pulling in transitive dependencies that are not needed.

```
10:31 $ jar tvf target/presto-verifier-0.265-SNAPSHOT-executable.jar | wc -l
   54540
 ```
   
I think adding the exclusion can be a good short term solution and the long term solution could be spinning out the metadata package (which looks like the main dependency) from presto-main as a separate module.

@rongrong / @tdcmeehan WDYT ?
